backend/tests/regression/answer_quality/relari.py

```diff
@@ -1,20 +1,53 @@
 import argparse
 import json
+import os
+import time
 
+import yaml
+from sqlalchemy import select
 from sqlalchemy.orm import Session
 
 from danswer.configs.constants import MessageType
-from danswer.db.engine import get_sqlalchemy_engine
+from danswer.db.engine import get_sqlalchemy_engine_for_port_number
+from danswer.db.index_attempt import get_inprogress_index_attempts
+from danswer.db.models import Document
 from danswer.one_shot_answer.answer_question import get_search_answer
 from danswer.one_shot_answer.models import DirectQARequest
 from danswer.one_shot_answer.models import OneShotQAResponse
 from danswer.one_shot_answer.models import ThreadMessage
 from danswer.search.models import IndexFilters
 from danswer.search.models import OptionalSearchSetting
 from danswer.search.models import RetrievalDetails
+from tests.regression.answer_quality.cli_utils import get_current_commit_sha
+from tests.regression.answer_quality.cli_utils import get_server_host_port
+
+
+def _get_answer_for_question(
+    query: str, db_session: Session, run_suffix: str
+) -> OneShotQAResponse:
+    while True:
```

Comment:
Should probably be behind a flag (ie this function should take a bool option) whether or not to assume that there can't be any ongoing indexing. But this is really nit and unimportant